### ISSUE TITLE
aws-ebs-csi-driver not using service account

### ISSUE THREAD

**Issue by reyyzzy** [kubernetes-sigs/aws-ebs-csi-driver#1717]
/kind bug

**What happened?**
i don't know why my aws-ebs-csi driver won't using the service account, i'm creating my service account using eksctl, i already annotate the service account and restart the deployment, and yes the role is exist, i already set the `automountServiceAccountToken: true` why is this happening?

what i've tried: 
reinstall the ebs-csi driver and recreate the iam roles

**What you expected to happen?**
ebs-csi driver using the service account

**How to reproduce it (as minimally and precisely as possible)?**
1. create iam service account
2. deploy aws-ebs-csi driver

**Anything else we need to know?**:

**Environment**
- Kubernetes version (use `kubectl version`): v1.25.12-eks-2d98532
- Driver version:2.21.0

**Comment by ConnorJC3** [comment]
Hi @reyyzzy, can you please elaborate on a few things:

1. How are you providing credentials to the EBS CSI Driver (IRSA, credentials secret, etc)?

2. What do you mean when you say the EBS CSI Driver "isn't using the service account"

3. Please provide any relevant log snippets of your configuration and/or this behavior

**Comment by reyyzzy** [comment]
> Hi @reyyzzy, can you please elaborate on a few things:
> 
> 1. How are you providing credentials to the EBS CSI Driver (IRSA, credentials secret, etc)?
> 2. What do you mean when you say the EBS CSI Driver "isn't using the service account"
> 3. Please provide any relevant log snippets of your configuration and/or this behavior

1. using eksctl
2. yes the EBS CSI Driver isn't using the service account, so i was following install steps like setup credentials, IAM role, etc and then install the driver, right now im changing to the add-on AWS EBS-CSI Driver, the Role that i specified in service account is exist and attached to _AmazonEBSCSIDriverPolicy_ but when i take. a look at the ebs-csi-controller-xxx Pod logs, this error show up 

`1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout`

im not sure why but when i take a look to the IAM Role that  i use as service account the Last Activity is `None` which means the ebs-csi-controller deployment is not using service account

3.  
```

E0815 07:21:18.431720       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:21:57.901500       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:22:36.964592       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:23:16.933254       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:23:54.441197       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:24:30.282672       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:25:07.865210       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:25:47.770348       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:26:25.038563       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:27:01.784384       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:27:41.099642       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:28:22.052738       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:29:01.048669       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:29:39.748734       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:30:16.684352       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:30:56.759639       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:31:33.134301       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:32:13.372991       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:32:49.115466       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:33:28.423899       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:34:04.262319       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:34:43.981161       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:35:21.061279       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:35:59.324993       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
E0815 07:36:37.878445       1 leaderelection.go:327] error retrieving resource lock kube-system/ebs-csi-aws-com: Get "https://172.20.0.1:443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/ebs-csi-aws-com": dial tcp 172.20.0.1:443: i/o timeout
```

**Comment by AndrewSirenko** [comment]
Hi @ConnorJC3 @reyyzzy, based on the above conversation I am going to mark this issue as closed for now. Feel free to reopen it if you are still running into EBS CSI Driver issues.

However, you running into this issue might point to our installation docs as needing increased clarity. If you have any suggestions for what can be made clearer do let me know. Thank you!

/close

**Comment by Atinlay** [comment]
The answers to this thread helped me get on the right track. I'm using the cilium CNI plugin which hands out IP's to pods from its own address space. The ebs-csi-driver-controller was deployed before the cilium CNI plugin, and most likely had an old IP cached to connect to the `kube-system/ebs-csi-aws-com`, resulting in a timeout. All I had to do was recreate the ebs-csi-driver-controller pods / perform a rolling restart of the ebs-csi-driver-controller deployment and it was fixed.
